substrait/function/FunctionLookup.cpp

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "substrait/function/FunctionLookup.h"
+
+namespace io::substrait {
+
+FunctionVariantPtr FunctionLookup::lookupFunction(
+    const FunctionSignature& signature) const {
+  const auto& functionMappings = getFunctionMap();
+
+  const auto& substraitFunctionName =
+      functionMappings.find(signature.name) != functionMappings.end()
+      ? functionMappings.at(signature.name)
+      : signature.name;
```

Comment:
I don't understand the purpose of `functionMappings`.  It appears to be allowing the user to override substrait names.  Why couldn't the user do this themselves before they call `lookupFunction`?

If I am using a substrait library I would expect it to use substrait names.  I know many engines have different names for functions but that does not seem to be a problem this library should be trying to solve.

Reply:
updated